import pprint
import time
import json
import psycopg2
from SECRETS import SECRETS

if __name__ == "__main__":

    try:
        connection_string = []
        connection_string.append("host=rhetoracle-db-instance.c2vrlkt9v1tp.us-west-2.rds.amazonaws.com")
        connection_string.append("dbname=rhetorical-db")
        connection_string.append("user=" + SECRETS['DB_USERNAME'])
        connection_string.append("password=" + SECRETS['DB_PASSWORD'])
        conn = psycopg2.connect(" ".join(connection_string))
    except Exception as x:
        print "Error connecting to DB: ", x.args

    cur = conn.cursor()

    tweet_id = "01234598344983493923940446"
    text = "Hey man this is a message"
    hashtags = "{hashtag1, hashtag2}"
    user_mentions = "{user-mention}"
    created_at = "Tue Jun 24 21:33:24 +0000 2014"
    screen_name = "FatTony"
    urls = "{http://www.website.com}"
    location = """["44", "55"]"""
    in_reply_to_screen_name = "pokerface"
    retweets = 99

    PUSH_SQL = """
    INSERT INTO massive(
        tweet_id, text, hashtags, user_mentions, created_at,
        screen_name, urls, location,
        inreplytostatusif, retweetcount)

    VALUES(
        '{}', '{}', '{}', '{}', '{}', '{}', '{}',
        '{}', '{}', {}); """

    test_sql = PUSH_SQL.format(tweet_id, text, hashtags, user_mentions,
                               created_at, screen_name, urls, location,
                               in_reply_to_screen_name, retweets)

    try:
        cur.execute(test_sql)
    except psycopg2.Error as x:
        # this will catch any errors generated by the database
        print test_sql
        print "Oops there was a DB error!", x.args
    else:
        conn.commit()
